packages/nextjs/pages/example-zk/BirthDateSignature.tsx

```diff
@@ -44,38 +46,52 @@ export const BirthDateSignature = ({ aliceDefaultAge }: { aliceDefaultAge: numbe
 
   return (
     <>
-      <div className="flex flex-col justify-center items-center bg-red-50">
-        <p>This is as the role of the authority in town.</p>
-        <p>Generate proof-of-birthyear-signature here.</p>
-        <AddressInput
-          value={form.personEthereumAddress}
-          name="personEthereumAddress"
-          placeholder="Ethereum address of person to sign for"
-          onChange={(val: string) => setForm({ ...form, personEthereumAddress: val })}
-        />
-        <input
-          type="number"
-          placeholder="birth-year"
-          value={form.birthYear}
-          onChange={e => setForm({ ...form, birthYear: e.target.value as unknown as number })}
-        />
-        <input
-          type="text"
-          placeholder="super-secret-key-for-signing"
-          value={form.theMayorsSecretKey}
-          onChange={e => setForm({ ...form, theMayorsSecretKey: e.target.value })}
-        />
-        <button className="border-2 border-black" onClick={handleSubmission}>
-          Generate proof-of-birthyear-signature
-        </button>
-        {signedBirthYear && signerPublicKey && (
-          <div className="h-200 w-80 bg-green-50">
-            <p>proof-of-birthyear-signature</p>
-            <p className="break-words">{signedBirthYear}</p>
-            <p>proof-of-birthyear-publickey</p>
-            <p className="break-words">{signerPublicKey}</p>
+      <div className="card flex-shrink-0 w-full max-w-lg shadow-2xl bg-base-100">
+        <div className="card-body">
+          <div className="form-control">
+            <label className="label">
+              <span className="label-text">Enter your Ethereum address</span>
+            </label>
+            <AddressInput
+              value={form.personEthereumAddress}
+              name="personEthereumAddress"
+              placeholder="Ethereum address"
+              onChange={(val: string) => setForm({ ...form, personEthereumAddress: val })}
+            />
+          </div>
+          <div className="form-control">
+            <label className="label">
+              <span className="label-text">Enter your birth year</span>
+            </label>
+            <input
+              type="number"
+              placeholder="Birth year"
+              className="input input-bordered"
+              value={form.birthYear}
+              onChange={e => setForm({ ...form, birthYear: e.target.value as unknown as number })}
```

Comment:
Do we really need to do this double casting for the `target.value`?

Reply:
will check

Reply:
removed the casting and replaced it with `Number()`, due to the state needing a number and the input always giving string as values